### ISSUE TITLE
Running consul as window service, the process is getting terminated

### ISSUE THREAD

**Issue by mukeshcs** [hashicorp/consul#5446]
When filing a bug, please include the following headings if possible. Any example text in this template can be deleted.

when i am creating consul as window service. Windows service getting created but it not running. we are getting error the process is getting terminated

A paragraph or two about the issue you're experiencing.

#### Reproduction Steps

Steps to reproduce this issue, eg:

Windows service created using below command.
sc create Consul binpath= "C:\Program Files\Consul\consul.exe agent -config-dir C:\Program Files\Consul\Config"

Service getting created but when i start it i get ant error.
service terminated unexpectedly. 

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

```
output from client 'consul info' command here
```

</details>

<details>
  <summary>Server info</summary>

```
output from server 'consul info' command here
```

</details>

### Operating system and Environment details

OS, Architecture, and any other information you can provide about the environment.

### Log Fragments

The Consul service terminated unexpectedly.  It has done this 2 time(s).

**Comment by pierresouchay** [comment]
@mukeshcs which version? Which version of windows? 
Don't you have any log?

**Comment by mukeshcs** [comment]
Windows : Windows 10 Enterprise 
consul version: v1.4.3

when i start the service i just see attachec message in event log

![consulog](https://user-images.githubusercontent.com/48345579/54017094-fd370d00-41aa-11e9-9f5d-00d5e3b678ab.JPG)

**Comment by mukeshcs** [comment]
We are facing this issue on windows 7 as well. My colleague tried ruining the same on window 7 but no luck.
Can you please let us know if it is possible to run consul as windows service or not. Or this is an issue from consul itself.

**Comment by avistyx** [comment]
+1

**Comment by pierresouchay** [comment]
@mukeshcs @avistyx Is it related to https://github.com/hashicorp/consul/pull/5501?

**Comment by judwhite** [comment]
@mukeshcs @avistyx Try specifying `--data-dir`.

My service creation script looks like this:

```
sc create consul binpath= "e:\consul\consul.exe agent --data-dir e:\consul\data --server --client 0.0.0.0" start= auto
sc failure consul actions= restart/5000/restart/60000/restart/300000// reset= 120
net start consul
```

**Comment by pearkes** [comment]
Given we haven't heard anything based on the suggestions/questions above I'm going to close this. Notably, we need to logs to help track down this issue. 

Feel free to open a new issue with the bug report template and provide the necessary information and we should be able to help track it down if you're still blocked on this.

**Comment by WuAiDeng** [comment]
+1

**Comment by liwsakilive** [comment]
+1

**Comment by WuAiDeng** [comment]
> +1

I can

**Comment by te4336** [comment]
@pearkes The service needs to start before it will create logs so you closed this somewhat prematurely. We've been battling this as well. Would also help if there was more comprehensive documentation surrounding Windows than the few snippets that are currently in there.
